org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/xbase/XbaseUsageDetector.java

```diff
@@ -35,7 +35,7 @@ public boolean usesXImportSection(Grammar grammar) {
 		new UsedRulesFinder(usedRules).compute(grammar);
 		for (AbstractRule it : usedRules) {
 			if ("XImportSection".equals(it.getName())
-					&& "org.eclipse.xtext.xbase.Xtype".equals(GrammarUtil.getGrammar(it).getName())) {
+					&& "http://www.eclipse.org/xtext/xbase/Xtype".equals(it.getType().getMetamodel().getEPackage().getNsURI()) {
```

Comment:
am not sure if this is null safe enhough

Reply:
can you also fix the compile error?

Reply:
Fixed the compile error.
I am also not sure if the chain is safe enough. I think getType can return null, getMetamodel and getEPackage will not return null, but can return a proxy. Shall I modify the code to check for proxies and null?

Reply:
can you test with something broken. e.g

XImportSection returns does::NotExist

Reply:
I am sorry, but I do not know how to do that, If I run the workflow which such a grammar, I get

XtextLinkingDiagnostic: null:23 Couldn't resolve reference to EClassifier 'NotExist'.
TransformationDiagnostic: null:23 Cannot create type 'NotExist' in alias null (ErrorCode: CannotCreateTypeInSealedMetamodel)
	at org.eclipse.xtext.xtext.generator.XtextGeneratorLanguage.initialize(XtextGeneratorLanguage.java:223)
	at org.eclipse.xtext.xtext.generator.XtextGenerator.initialize(XtextGenerator.java:175)
	at org.eclipse.xtext.xtext.generator.XtextGenerator.checkConfigurationInternal(XtextGenerator.java:129)
	at org.eclipse.emf.mwe.core.lib.AbstractWorkflowComponent2.checkConfiguration(AbstractWorkflowComponent2.java:23)
	at org.eclipse.emf.mwe.core.lib.Mwe2Bridge.preInvoke(Mwe2Bridge.java:69)
	at org.eclipse.emf.mwe.core.lib.AbstractWorkflowComponent.preInvoke(AbstractWorkflowComponent.java:208)
	at org.eclipse.emf.mwe2.runtime.workflow.AbstractCompositeWorkflowComponent.preInvoke(AbstractCompositeWorkflowComponent.java:31)
	at org.eclipse.emf.mwe2.runtime.workflow.Workflow.run(Workflow.java:20)
	at org.eclipse.emf.mwe2.launch.runtime.Mwe2Runner.run(Mwe2Runner.java:103)
	... 4 more

so I do not get to the point where the is run and XbaseGeneratorFragment2 is run as well.

Reply:
so it looks like it practially cant get null
@szarnekow what do you think

Reply:
Yes, I think in practice it cannot. 

I could also contribute our test for the fix, which would be in 
https://github.com/dsldevkit/dsl-devkit/blob/351f81c0a5a7d78f0ae0bfe78ab19bc42cc46c62/com.avaloq.tools.ddk.xtext.generator.test/src/com/avaloq/tools/ddk/xtext/generator/xbase/test/FixedXbaseGeneratorFragmentTest.xtend

but I am unsure how your test setup works. Could I put a similar class in org.eclipse.xtext.tests/generator/src/org/eclipse/xtext)/tests and it will be executed by your lunch config [xtext.tests (xtend).launch](https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.tests/xtext.tests%20(xtend).launch)?